pkg/common/keys.go

```diff
@@ -160,4 +160,24 @@ const (
 
 	// ArgoCDKeyUsersAnonymousEnabled is the configuration key for anonymous user access.
 	ArgoCDKeyUsersAnonymousEnabled = "users.anonymous.enabled"
+
+	// ArgoCDDexImageEnvName is the environment variable used to get the image
+	// to used for the Dex container.
+	ArgoCDDexImageEnvName = "ARGOCD_DEX_IMAGE"
```

Comment:
Do we need to document these environment variables possibly in docs/reference/argocd.md?

Reply:
Good point, I guess I'd originally assumed we'd do that as part of the image declarations in the yaml, but we can do it here.

How do you think we should call this out, with a table at the start, with a list of keys (within the CRD) and their equivalent environment-variables? Or should each image (they are spread out through the document) indicate that the YAML overrides the environment variable?

Reply:
I like the latter a bit more.   For example, https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#dex-options here we can add documentation in the description of `Image` to state something like the default is overridden by `ARGOCD_DEX_IMAGE`  environment variable.    I think it is also helpful to document how/where to set an environment.